crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -590,6 +609,22 @@ impl Entities {
         self.meta.get_unchecked_mut(index as usize).location = location;
     }
 
+    /// Increments the generation of a despawned [`Entity`]. This may only be called on an entity
+    /// index corresponding to an entity that once existed and has since been freed.
+    pub(crate) fn reserve_generations(&mut self, index: u32, generations: u32) -> bool {
```

Comment:
Maybe keep the convention of prefixing methods returning a boolean result with `try_`?

Reply:
Is that a convention bevy holds to? Right now we have `World::despawn` instead of `World::try_despawn`, for instance.

Reply:
Only loosely :( It's a good convention, although I like it even better when it returns results.

Reply:
Yes, it's also present in the standard library, and in other languages like C#. I generally like using it if I can to make the returned `bool` a bit more self-documented.

Reply:
That's a minor suggestion though feel free to ignore.

Reply:
I was copying the pattern in `despawn` since `reserve_generations` is sort of an extension of it. I'll maybe switch this to returning a result or at least rename it.